src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotPopAndLockDatasetStep.java

```diff
@@ -78,11 +78,9 @@ public StepResult doStep(FlightContext context) {
 
   @Override
   public StepResult undoStep(FlightContext context) {
+    FlightMap workingMap = context.getWorkingMap();
     try {
-      UUID sourceDatasetId =
-          snapshotService.getFirstSourceDatasetIdFromSnapshotId(
```

Comment:
This appears to be the only use of `SnapshotService.getFirstSourceDatasetIdFromSnapshotId()`. If so, you should delete it from `SnapshotService` as well.